### PR TITLE
Added support for producing TS files when creating a new package.

### LINKED ISSUE #102: Support for producing TS files when creating a new package
### Story

- **Given**: I am a user and want to have `.ts` files when creating a new package.
- **When**: I execute the `create-package ...` command.
- **Then**: The tool should ask about the output type that allows choosing between JavaScript (default) and TypeScript.

### Hints

- CKEditor 5, so far, does not have published typings. Hence, we need to use typings produced by the community: `@types/ckeditor__ckeditor5-*`
- We need to figure out how to maintain source files. Do we want a single file that will be aligned based on the target language?
- A TypeScript project might need additional files, e.g., `tsconfig.json`.

### PR DESCRIPTION
**Pull request by przemyslaw-zan**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Added support for producing TS files when creating a new package. Closes #102.

---

### Additional information

*For example – encountered issues, assumptions you had to make, other affected tickets, etc.*

### Supported scopes

* `generator` → https://www.npmjs.com/package/ckeditor5-package-generator
* `*` → https://www.npmjs.com/package/@ckeditor/ckeditor5-package-*

### CODE REVIEW

File: packages/ckeditor5-package-generator/lib/templates/ts/package.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    "lint-staged": "^10.2.6",
+    "stylelint": "^13.13.1",
+    "stylelint-config-ckeditor5": ">=<%= packageVersions.stylelintConfigCkeditor5 %>",
+    "ts-node": "^10.9.1",
+    "typescript": "^4.7.4"
+  },
+  "scripts": {
+    "dll:build": "echo \"TODO: DLL in #103\"",
+    "dll:serve": "http-server ./ -o sample/dll.html",
+    "lint": "eslint \"**/*.js\" --quiet --ignore-pattern \"build/\"",
```
Comment (psmyrek) [inline packages/ckeditor5-package-generator/lib/templates/ts/package.json:68]: In the generated TypeScript package we should definitely lint `.ts` files.

```suggestion
    "lint": "eslint \"**/*.ts\" --quiet --ignore-pattern \"build/\"",
```

File: packages/ckeditor5-package-generator/lib/templates/ts/package.json [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    "stylelint": "stylelint --quiet --allow-empty-input 'theme/**/*.css'",
+    "test": "echo \"TODO: DLL in #104\"",
+    "prepare": "<%= program %> run dll:build",
+    "prepublishOnly": "<%= program %> run dll:build",
+    "translations:collect": "ckeditor5-package-tools translations:collect",
+    "translations:download": "ckeditor5-package-tools translations:download",
+    "translations:upload": "ckeditor5-package-tools translations:upload"
+  },
+  "lint-staged": {
+    "**/*.js": [
```
Comment (psmyrek) [inline packages/ckeditor5-package-generator/lib/templates/ts/package.json:79]:
```suggestion
    "**/*.{js,ts}": [
```

**Review by psmyrek — COMMENTED** [review]
It works nice 🎉 

Some files in the `packages/ckeditor5-package-generator/lib/templates/common/` I think should also be split into JavaScript and TypeScript versions:

1. `ckeditor5-metadata.json` contains `"path": "src/myplugin.js"`, but for TypeScript version it should point to the `src/myplugin.ts`.
2. `.gitattributes` should contain an entry for `*.ts` files.
3. `sample/index.html` contains the editor content with folder structure for the JavaScript version.

The `README.md` file in the `lint` section contains a note about linting `*.js` files:

> Runs ESLint, which analyzes the code (all `*.js` files) to quickly find problems.

To not split it into JavaScript and TypeScript versions, let's simplify it by removing the `(all ``*.js`` files)`.

**Comment by przemyslaw-zan** [comment]
> I did not remove this comment that was in the code. Most of the points seem no longer relevant. The TODO part should be a ticket instead of a comment.
> 
> https://github.com/ckeditor/ckeditor5-package-generator/blob/70f8fa05ec875ca63e7e428e869e4ef9bb3e359f/packages/ckeditor5-package-generator/lib/index.js#L27-L34

After a discussion, we've decided that the `--info` flag is not needed, and that the points are no longer relevant.

**Review by psmyrek — APPROVED** [review]
LGTM 🎉

**Pull request merged**